elastic/apm-server
# config: Remove deprecated settings for `8.0`

## Motivation/summary

Removes all the deprecated config settings, updating the unit and system
tests to use the new non deprecated fields.

## Checklist


## How to test these changes

1. `docker-compose up -d`
2. `make`
3. Run apm server and verify that `ignoring deprecated config` isn't found in the logs:
```console
$ ./apm-server -e -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -E apm-server.instrumentation.enabled=true \
  -E apm-server.rum.allowed_service=\[aservice\] \
  -E apm-server.rum.event_rate.limit=1 \
  -E apm-server.api_key.enabled=true \
  -E apm-server.secret_token=some_secret
```
4. `curl localhost:8200/foo`
5. Via Kibana dev tools verify there are no instrumentation metrics / traces ingested:
```javascript
GET apm-*,traces-apm*,logs-apm*/_search
{
  "query": {
    "match_all": {}
  }
}
```

## Related issues

Closes #5996
